src/optimizer/join_order/relation_statistics_helper.cpp

```diff
@@ -0,0 +1,348 @@
+
+struct DistinctCount;
+
+static ExpressionBinding GetChildColumnBinding(Expression *expr) {
+	auto ret = ExpressionBinding();
+	switch (expr->expression_class) {
+	case ExpressionClass::BOUND_FUNCTION: {
+		// TODO: Other expression classes that can have 0 children?
+		auto &func = expr->Cast<BoundFunctionExpression>();
+		// no children some sort of gen_random_uuid() or equivalent.
+		if (func.children.size() == 0) {
+			ret.found_expression = true;
+			ret.expression_is_constant = true;
+			return ret;
+		}
+		break;
+	}
+	case ExpressionClass::BOUND_COLUMN_REF: {
+		ret.found_expression = true;
+		auto &new_col_ref = expr->Cast<BoundColumnRefExpression>();
+		ret.child_binding = ColumnBinding(new_col_ref.binding.table_index, new_col_ref.binding.column_index);
+		return ret;
+	}
+	case ExpressionClass::BOUND_LAMBDA_REF:
+	case ExpressionClass::BOUND_CONSTANT:
+	case ExpressionClass::BOUND_DEFAULT:
+	case ExpressionClass::BOUND_PARAMETER:
+	case ExpressionClass::BOUND_REF:
+		ret.found_expression = true;
+		ret.expression_is_constant = true;
+		return ret;
+	default:
+		break;
+	}
+	ExpressionIterator::EnumerateChildren(*expr, [&](unique_ptr<Expression> &child) {
+		auto recursive_result = GetChildColumnBinding(child.get());
+		if (recursive_result.found_expression) {
+			ret = recursive_result;
+		}
+	});
+	// we didn't find a Bound Column Ref
+	return ret;
+}
+
+RelationStats RelationStatisticsHelper::ExtractGetStats(LogicalGet &get, ClientContext &context) {
+	auto return_stats = RelationStats();
+
+	auto base_table_cardinality = get.EstimateCardinality(context);
+	auto cardinality_after_filters = base_table_cardinality;
+	unique_ptr<BaseStatistics> column_statistics;
+
+	auto table_thing = get.GetTable();
+	auto name = string("some table");
+	if (table_thing) {
+		name = table_thing->name;
+		return_stats.table_name = name;
+	}
+	// first push back basic distinct counts for each column (if we have them).
+	for (idx_t i = 0; i < get.column_ids.size(); i++) {
+		bool have_distinct_count_stats = false;
+		if (get.function.statistics) {
+			column_statistics = get.function.statistics(context, get.bind_data.get(), get.column_ids[i]);
+			if (column_statistics) {
+				auto column_distinct_count = DistinctCount({column_statistics->GetDistinctCount(), true});
+				return_stats.column_distinct_count.push_back(column_distinct_count);
+				return_stats.column_names.push_back(name + "." + get.names.at(get.column_ids.at(i)));
+				have_distinct_count_stats = true;
+			}
+		}
+		if (!have_distinct_count_stats) {
+			// TODO: currently treating the cardinality as the distinct count.
+			//  the cardinality estimator will update these distinct counts based
+			//  on the extra columns that are joined on.
+			// TODO: Should this be changed?
+			auto column_distinct_count = DistinctCount({cardinality_after_filters, false});
+			return_stats.column_distinct_count.push_back(column_distinct_count);
+			// TODO: we can still get parquet column names right?
```

Comment:
Yes, we can get parquet column names. It should be in the `LogicalGet` just like for tables